sdks/python/apache_beam/io/range_trackers.py

```diff
@@ -101,11 +106,45 @@ def _validate_record_start(self, record_start, split_point):
 
   def try_claim(self, record_start):
     with self._lock:
+      if self.done():
+        raise ValueError('Trying to claim record at %r after already closing '
+                         'RangeTracker by invoking \'set_done()\'',
+                         record_start)
+
       self._validate_record_start(record_start, True)
       if record_start >= self.stop_position():
         return False
       self._offset_of_last_split_point = record_start
       self._last_record_start = record_start
+      self._split_points_seen += 1
+      if self._split_points_remaining_callback:
+        if (self._split_points_remaining ==
+            iobase.RangeTracker.SPLIT_POINTS_UNKNOWN):
+          if not self._first_claim_done:
+            # Trying to set the initial value for remaining number of split
+            # points.
+            # We only try to do this at first claim and when stop position gets
+            # updated so that we do not unnecessarily keep calling the callback
```

Comment:
try_claim is called quite frequently; we should only invoke the callback when the reporter thread asks for the number of split points. Here all we should do is increment _split_points_seen.